diff --git a/envs/_base_task.py b/envs/_base_task.py
index 982e6ec..f6aba39 100644
--- a/envs/_base_task.py
+++ b/envs/_base_task.py
@@ -207,7 +207,7 @@ class Base_Task(gym.Env):
         from sapien.render import set_global_config
 
         set_global_config(max_num_materials=50000, max_num_textures=50000)
-        self.renderer = sapien.SapienRenderer()
+        self.renderer = sapien.render.SapienRenderer(sapien.Device("cuda:0"))
         # give renderer to sapien sim
         self.engine.set_renderer(self.renderer)
 
@@ -218,7 +218,11 @@ class Base_Task(gym.Env):
 
         # declare sapien scene
         scene_config = sapien.SceneConfig()
-        self.scene = self.engine.create_scene(scene_config)
+        sapien.physx.set_scene_config(scene_config)
+        self.scene = sapien.Scene([
+            sapien.physx.PhysxCpuSystem(),
+            sapien.render.RenderSystem(sapien.Device("cuda:0")),
+        ])
         # set simulation timestep
         self.scene.set_timestep(kwargs.get("timestep", 1 / 250))
         # add ground to scene
diff --git a/script/test_render.py b/script/test_render.py
index 97350aa..ed49b35 100644
--- a/script/test_render.py
+++ b/script/test_render.py
@@ -63,7 +63,7 @@ class Sapien_TEST(gym.Env):
         from sapien.render import set_global_config
 
         set_global_config(max_num_materials=50000, max_num_textures=50000)
-        self.renderer = sapien.SapienRenderer()
+        self.renderer = sapien.render.SapienRenderer(sapien.Device("cuda:0"))
         # give renderer to sapien sim
         self.engine.set_renderer(self.renderer)
 
@@ -74,7 +74,11 @@ class Sapien_TEST(gym.Env):
 
         # declare sapien scene
         scene_config = sapien.SceneConfig()
-        self.scene = self.engine.create_scene(scene_config)
+        sapien.physx.set_scene_config(scene_config)
+        self.scene = sapien.Scene([
+            sapien.physx.PhysxCpuSystem(),
+            sapien.render.RenderSystem(sapien.Device("cuda:0")),
+        ])
 
 
 if __name__ == "__main__":
